packages/neuron-wallet/src/models/synced-block-number.ts

```diff
@@ -4,64 +4,54 @@ import SyncedBlockNumberSubject from "models/subjects/node"
 
 // Keep track of synced block number.
 export default class SyncedBlockNumber {
-  private nextBlock: bigint | undefined = undefined
+  #blockNumberEntity: SyncInfoEntity | undefined = undefined
+  #nextBlock: bigint | undefined = undefined
+  #lastSavedBlock: bigint = BigInt(-1)
 
   // Get next block to scan. If syncing hasn't run yet return 0 (genesis block number).
   public getNextBlock = async (): Promise<bigint> => {
-    if (this.nextBlock) {
-      return this.nextBlock
+    if (this.#nextBlock) {
+      return this.#nextBlock
     }
 
-    const blockNumberEntity: SyncInfoEntity | undefined = await this.blockNumber()
-
-    if (!blockNumberEntity) {
-      return BigInt(0)
-    }
-
-    return BigInt(blockNumberEntity.value)
+    return BigInt((await this.blockNumber()).value)
   }
 
   public setNextBlock = async (current: bigint): Promise<void> => {
-    this.nextBlock = current
+    this.#nextBlock = current
     SyncedBlockNumberSubject.getSubject().next(current.toString())
 
-    let blockNumberEntity = await this.blockNumber()
-
-    let skipSave = false
-    if (blockNumberEntity && current - BigInt(blockNumberEntity.value) < BigInt(1000)) {
+    if (current === BigInt(0) || this.#lastSavedBlock === BigInt(-1) || current - this.#lastSavedBlock >= BigInt(1000)) {
       // Only persist block number for every 1,000 blocks to reduce DB write.
       // Practically it's unnecessary to save every block height, as iterating
       // blocks is fast.
-      skipSave = true
-    }
-    if (current === BigInt(0)) {
-      // Initial or reset
-      skipSave = false
-    }
 
-    if (skipSave) {
-      return
-    }
+      this.#lastSavedBlock = current
 
-    if (!blockNumberEntity) {
-      blockNumberEntity = new SyncInfoEntity()
-      blockNumberEntity.name = SyncInfoEntity.CURRENT_BLOCK_NUMBER
+      let blockNumberEntity = await this.blockNumber()
+      blockNumberEntity.value = current.toString()
+      getConnection().manager.save(blockNumberEntity)
```

Comment:
`await` ?

Reply:
This is by purpose. It doesn't need to wait for the writing to be finished.

Not sure if this would cause any side effect, though.

Reply:
OK